src/components/NewCommentForm.tsx

```diff
@@ -1,37 +1,192 @@
-import React from 'react';
+import React, { useState } from 'react';
+import cn from 'classnames';
+import { CommentData } from '../types/Comment';
+import { ErrorMessage } from './ErrorMessage';
+
+interface ErrorMessages {
+  nameError: string,
+  emailError: string,
+  bodyError: string,
+  serverError: string,
+}
+
+const defaultErrorValues = {
+  nameError: '',
+  emailError: '',
+  bodyError: '',
+  serverError: '',
+};
+
+interface Props {
+  handleAddNewComment: (commentData: CommentData) => Promise<void>,
+}
+
+export const NewCommentForm: React.FC<Props> = ({
+  handleAddNewComment,
+}) => {
+  const [errors, setErrors] = useState<ErrorMessages>(defaultErrorValues);
+
+  const [commentData, setCommentData] = useState<CommentData>({
+    name: '',
+    email: '',
+    body: '',
+  });
+
+  const [isLoading, setIsLoading] = useState(false);
+
+  const handleReset = () => {
+    setCommentData({
+      name: '',
+      email: '',
+      body: '',
+    });
+    setErrors({
+      ...errors,
+      nameError: '',
+      emailError: '',
+      bodyError: '',
+    });
+  };
+
+  const handleSubmit = (event: React.FormEvent<HTMLFormElement>) => {
+    event.preventDefault();
+
+    const normalizedName = commentData.name.trim();
+    const normalizedEmail = commentData.email.trim();
+    const normalizedBody = commentData.body.trim();
+
+    let nameError = '';
+    let emailError = '';
+    let bodyError = '';
+
+    if (!normalizedName) {
+      nameError = 'Name is required';
+    }
+
+    if (!normalizedEmail) {
+      emailError = 'Email is required';
+    }
+
+    if (!normalizedBody) {
+      bodyError = 'Enter some text';
+    }
+
+    if (nameError || emailError || bodyError) {
+      setErrors({
+        ...errors,
+        nameError,
+        emailError,
+        bodyError,
+      });
+
+      return;
+    }
+
+    setIsLoading(true);
+
+    handleAddNewComment({
+      name: normalizedName,
+      email: normalizedEmail,
+      body: normalizedBody,
+    })
+      .then(() => {
+        setErrors(defaultErrorValues);
+
+        setCommentData({
+          ...commentData,
+          body: '',
+        });
+      })
+      .catch(() => {
+        setErrors({
+          ...errors,
+          serverError: 'Unable to add a comment',
+        });
+      })
+      .finally(() => setIsLoading(false));
+  };
+
+  const handleNameChanged = (event: React.ChangeEvent<HTMLInputElement>) => {
+    setErrors({
+      ...errors,
+      nameError: '',
+    });
+
+    setCommentData({
+      ...commentData,
+      name: event.target.value,
```

Comment:
You can accept key and then use {
      ...commentData,
      [key]: event.target.value,
      }
then count of handlers be minimize ( actual for switch cases under)